course-content/32_graphical_depictions_of_data.qmd

```diff
@@ -0,0 +1,669 @@
+    kind='bar', hue='sex', estimator=np.sum)
+plt.show()
+```
+:::
+
+One last unique piece that you might be interested in is the orientation of the plot. In the above examples, we gave our categorical category to the `x` feature, and `y` held the numerical values. If we wish to have a horizontal set of bars instead, we can simply swap these:
+
+```{python}
+sns.catplot(
+    data=penguins, x='flipper_length_mm', y='species',
+    kind='bar', hue='sex')
+plt.show()
+```
+
+
+#### Summary
+To summarize the categorical plots available in Seaborn:
+
+* `catplot` is the go-to tool to create a statistical comparison plot between a numerical and categorical column
+* These plots default to categorical swarmplots (*no, I do not know why bar is not the default*) but other options are available. These options can be organized into three broad subcategories. 
+    * Categorical scatterplots:
+        
+         * `swarm`
+         * `strip`
+    * Categorical distribution plots:
+
+         * `box`
+         * `violin`
+         * `boxen`
+    * Categorical estimate plots:
+         * `point`
+         * `bar`
+         * `count`
+* `hue` can be adjusted to provide insight into extra categorical features
+* `row` and `col` will split our plot out into multiple facets, according to the categories found in the datasets
+* By swapping the `x` and `y` variables, we can choose which orientation the plot will display in
+
+:::{.callout-warning icon="false"}
+## Challenge 4
+
+Which penguin species was most frequently included in this dataset? Display your result visually!  
+:::
+
+:::{.callout-note icon="false" collapse="true"}
+## Solution to Challenge 4
+
+To get the count for each of the species, we need to make a countplot:
+
+```{python}
+sns.catplot(data=penguins, x='species', kind='count')
+plt.show()
+```
+
+Done, in one line of code!
+
+As a bonus, if we want to plot the data in descending order, we can pass a list of strings containing the categories to the `catplot` method:
+
+```{python}
+sns.catplot(
+    data=penguins, x='species', 
+    kind='count', 
+    order=['Adelie', 'Gentoo', 'Chinstrap']
+    )
+plt.show()
+```
+:::
+
+## Saving your Plots
+
+We've now done all this work to make amazing incredible charts that produce stunning visuals for our datasets. Chances are, however that you do not want these images to live solely in your Jupyter notebook. You might want it for a presentation or a report. So, we need to save these images so we can access them outside of our Python environment. To do this we use the `plt.savefig()` method. This method will save the **currently open figure**. This is the figure that has not yet been shown on screen! Any call to save a plot must come *before* we display it via `plt.show()`, as this actually closes the figure so we can start our next one!  
+
+:::{.callout-important}
+## No Show and Tell! 
+
+This point is important enough that I am repeating it here. Always save your figure **before** showing it! 
```

Comment:
Worth pointing out that this is because `show` clears the plotting device?

Reply:
Do I not say that somewhere? I thought I did in the matplotlib section... I can reiterate here however.

Reply:
Yup, you definitely do, just think it's worth pointing out here again